js/ui/codeView.js

```diff
@@ -96,8 +98,16 @@ const _ensureHackDataFile = (function () {
             return keyfile;
 
         try {
-            keyfile.load_from_dirs('hack-data.ini', searchPaths,
+            const [, path] = keyfile.load_from_dirs('hack-data.ini', searchPaths,
                 GLib.KeyFileFlags.NONE);
+            monitorFile = Gio.file_new_for_path(path);
+            monitor = monitorFile.monitor_file(Gio.FileMonitorFlags.NONE, null);
```

Comment:
If the config file is ever going to be modified from an OSTree commit, you might also want `Gio.FileMonitorFlags.WATCH_HARD_LINKS`, I suspect?

Reply:
done